common/tqid/task_queue_validator.go

```diff
@@ -67,14 +102,14 @@ func NormalizeAndValidate(
 		taskQueue.Name = defaultName
 	}
 
-	if err := Validate(taskQueue.GetName(), maxIDLengthLimit); err != nil {
+	if err := validate(taskQueue.GetName(), maxIDLengthLimit, expectNonRootPartition); err != nil {
 		return err
 	}
 
 	if taskQueue.GetKind() == enumspb.TASK_QUEUE_KIND_STICKY {
 		normalName := taskQueue.GetNormalName()
 		if normalName != "" {
```

Comment:
Not changing in this PR, but this might deserve a comment that old SDKs send sticky without a normal name, so we don't require that it be set here.